larq/utils.py

```diff
@@ -68,3 +70,17 @@ def patch_object(object, name, value):
     setattr(object, name, value)
     yield
     setattr(object, name, old_value)
+
+
+def _check_lce_version():
+    try:
+        import larq_compute_engine as lce  # pytype: disable=import-error
+
+        lce_version = getattr(lce, "__version__", "0.0.0")
+        if version.parse(lce_version) < version.parse("0.6.1"):
+            warnings.warn(
+                f"larq-compute-engine={lce_version} is not supported by this version of larq.\n"
+                "Please upgrade larq-compute-engine to v0.6.1 or newer."
+            )
+    except ImportError:
+        pass
```

Comment:
How do you feel about instead doing the version check of LCE using `importlib.metadata`, which would mean we wouldn't have to import LCE? Like we do for getting the Larq version itself:
https://github.com/larq/larq/blob/4ae5cc9e8f49a8db844e6e0c84d52c23eac10fee/larq/__init__.py#L15-L21

I only ask because at the moment Larq imports very quickly, whereas LCE imports very slowly (as slowly as TF does).

Alternatively we could do the check in `setup.py` as you say, though I think the problem there is that that means a warning would not be shown if first Larq is installed, and then LCE.

Reply:
Good idea! I thought the main slowness of the LCE import comes from TF itself, but it's good to just use `importlib.metadata` anyway since it's super fast.

Reply:
Good idea! I thought the main slowness of the LCE import comes from TF itself, but it's good to just use `importlib.metadata` anyway since it's super fast.